llarp/crypto/types.cpp

```diff
@@ -32,14 +33,44 @@ namespace llarp
     return oxenc::to_hex(begin(), end());
   }
 
+  PubKey::operator RouterID() const
+  {
+    return {as_array()};
+  }
+
+  PubKey&
+  PubKey::operator=(const byte_t* ptr)
+  {
+    std::copy(ptr, ptr + SIZE, begin());
+    return *this;
+  }
+
+  bool
+  operator==(const PubKey& lhs, const PubKey& rhs)
+  {
+    return lhs.as_array() == rhs.as_array();
+  }
+
+  bool
+  operator==(const PubKey& lhs, const RouterID& rhs)
+  {
+    return lhs.as_array() == rhs.as_array();
+  }
+
+  bool
+  operator==(const RouterID& lhs, const PubKey& rhs)
+  {
+    return lhs.as_array() == rhs.as_array();
+  }
```

Comment:
I'm pretty sure that we can get rid of both of these from here and the `.hpp` because the two-PubKey `operator==` should already properly match any pubkey+routerid pair (due to the added inheritance).